tests/v2/validation/resource/terraform/v2charts/provision/aws/main.tf

```diff
@@ -0,0 +1,294 @@
+#Cluster creation with v1 charts installed
+terraform {
+  required_version = ">= 0.14"
+  required_providers {
+    rancher2 = {
+      source = "rancher/rancher2"
+    }
+
+  }
+}
+
+provider "rancher2" {
+  api_url   = var.rancher_api_url
+  token_key = var.rancher_token_key
+}
+
+resource "rancher2_cluster" "v2charts" {
+  name = var.cluster_name
+  description = "Rancher v2 Charts Cluster"
+  rke_config {
+    kubernetes_version = var.k8s_version
+    network {
+      plugin = "canal"
+    }
+  }  
+}
+
+resource "rancher2_node_template" "node-template" {
+  name = "tf-aws-node-template"
+  description = "TF AWS node template"
+
+  amazonec2_config {
+    access_key = var.aws_access_key
+    secret_key = var.aws_secret_key
+    ami =  var.ami
+    ssh_user = var.ami_user
+    region = var.aws_region
+    security_group = var.security_groups
+    subnet_id = var.subnet
+    vpc_id = var.vpc_id
+    zone = var.zone
+    root_size = var.root_size
+    instance_type = var.instance_type
+  }
+}
+
+resource "rancher2_node_pool" "cp-node-pool" {
+  cluster_id =  rancher2_cluster.v2charts.id
+  name = "tf-cp-node-pool"
+  hostname_prefix = var.hostname_prefix_cp
+  node_template_id = rancher2_node_template.node-template.id
+  quantity = 2
+  control_plane = true
+  etcd = false
+  worker = false
+}
+
+resource "rancher2_node_pool" "etcd-node-pool" {
+  cluster_id =  rancher2_cluster.v2charts.id
+  name = "tf-etcd-node-pool"
+  hostname_prefix = var.hostname_prefix_etcd
+  node_template_id = rancher2_node_template.node-template.id
+  quantity = 3
+  control_plane = false
+  etcd = true
+  worker = false
+}
+
+resource "rancher2_node_pool" "worker-node-pool" {
+  cluster_id =  rancher2_cluster.v2charts.id
+  name = "tf-worker-node-pool"
+  hostname_prefix = var.hostname_prefix_worker
+  node_template_id = rancher2_node_template.node-template.id
+  quantity = var.worker_count
+  control_plane = false
+  etcd = false
+  worker = true
+}
+
+resource "rancher2_cluster_sync" "v2charts-sync" {
+  cluster_id = rancher2_cluster.v2charts.id
+  wait_catalogs = true
+  state_confirm = 90
+}
+
+resource "local_file" "sync_kube_config" {
+    depends_on = [rancher2_cluster_sync.v2charts-sync]
+    content  = rancher2_cluster_sync.v2charts-sync.kube_config
+    filename = "kube_config_provisioned_cluster.yaml"
+}
+
+resource "null_resource" "longhorn-iscsi-nfs" {
+  depends_on = [local_file.sync_kube_config]
+  provisioner "local-exec" {
+    environment = {
+      KUBECONFIG = "kube_config_provisioned_cluster.yaml"
+    }
+    command = "kubectl create namespace longhorn-system && kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/v1.3.0/deploy/prerequisite/longhorn-iscsi-installation.yaml --namespace=longhorn-system && kubectl apply -f https://raw.githubusercontent.com/longhorn/longhorn/v1.3.0/deploy/prerequisite/longhorn-nfs-installation.yaml --namespace=longhorn-system"
```

Comment:
Should this be hardcoded? Will we ever want to change the version?

Reply:
Added Longhorn prereq variable and set current defaults for 2.5 and 2.6
	- This removes the hardcoded version number